### PR TITLE
add rawtext syntax for lovelace and custom body-inset

### PR DESCRIPTION
**Pull request by OrangeX4**
https://github.com/andreasKroepelin/lovelace/issues/2

### CODE REVIEW

**Comment by OrangeX4** [comment]
@andreasKroepelin Any progress here?

File: lib.typ [hunk 1]
```diff
@@ -2,11 +2,44 @@
...
+  text(size: .7em, fill: gray, sym.triangle.stroked.r + sym.space + body)
+}
+
+#let _typst2lovelace(typst-code, scope: (:)) = {
+  let indent = 0
+  let last-indent = 0
+  let res = ()
+  for line in typst-code.text.split("\n") {
+    let whitespaces = line.find(regex("^\\s+"))
+    let current-indent = if (whitespaces != none) { whitespaces.len() } else { 0 }
```
Comment (andreasKroepelin) [inline lib.typ:16]:
```suggestion
    let current-indent = if whitespaces != none { whitespaces.len() } else { 0 }
```

File: lib.typ [hunk 2]
```diff
@@ -2,11 +2,44 @@
...
+    if (indent == 0 and current-indent != 0) {
```
Comment (andreasKroepelin) [inline lib.typ:17]:
```suggestion
    if indent == 0 and current-indent != 0 {
```

File: lib.typ [hunk 3]
```diff
@@ -2,11 +2,44 @@
...
+      indent = current-indent
+    }
+    if (current-indent > last-indent) {
```
Comment (andreasKroepelin) [inline lib.typ:20]:
```suggestion
    if current-indent > last-indent {
```

File: lib.typ [hunk 4]
```diff
@@ -2,11 +2,44 @@
...
+      res = res + (ind,) * int((current-indent - last-indent) / indent)
+    } else if (current-indent < last-indent) {
```
Comment (andreasKroepelin) [inline lib.typ:22]:
```suggestion
    } else if current-indent < last-indent {
```

File: lib.typ [hunk 5]
```diff
@@ -25,7 +58,16 @@
     x
   }
 
-  for child in children.pos() {
+  let children = ()
+  for child in args.pos() {
+    if type(child) == "content" and child.func() == raw {
```
Comment (andreasKroepelin) [inline lib.typ:63]:
```suggestion
    if type(child) == content and child.func() == raw {
```

File: lib.typ [hunk 6]
```diff
@@ -25,7 +58,16 @@
...
+      children = children + _typst2lovelace(child, scope: scope)
```
Comment (andreasKroepelin) [inline lib.typ:64]:
```suggestion
      children.push(_typst2lovelace(child, scope: scope))
```
for consistency with the other branch

**Review by andreasKroepelin — CHANGES_REQUESTED** [review]
Thanks for the PR, I made some suggestions regarding code style. Looks good otherwise!

File: README.md [hunk 7]
```diff
@@ -174,6 +215,7 @@ Also, there are some optional arguments to `lovelace-setup`:
   `#pseudocode` as the latter has an effect on line numbers in references as well.
 - `line-number-supplement`: some content that is placed before the line number
   when referencing it, default `"Line"`
+- `body-inset`: the inset of body, default to be `(bottom: 5pt)`
```
Comment (andreasKroepelin) [inline README.md:218]:
```suggestion
- `body-inset`: the inset of body, default `(bottom: 5pt)`
```

for consistency with the other bullet points

File: README.md [hunk 8]
```diff
@@ -131,6 +131,47 @@ However, Lovelace provides a sensible `#comment` function you can use:
...
+#show: setup-lovelace
+
+#let redbold(it) = {
+  set text(fill: red, weight: "bold")
+  it
+}
+
+#let pseudocode = pseudocode.with(scope: (redbold: redbold))
+
+#algorithm(
```
Comment (andreasKroepelin) [inline README.md:149]: Maybe actually don't use `#algorithm` here, it is not necessary for a minimal example.

**Review by andreasKroepelin — CHANGES_REQUESTED** [review]
Just some more things that came up.
- [ ] Also, please add an explanation for the `scope` keyword argument to the readme

**Comment by andreasKroepelin** [comment]
Alright, I resolved some code style issues and adopted a new API. Thanks again for your contribution!

**Comment by OrangeX4** [comment]
Thanks for review!

**Pull request merged**